keyboards/thevankeyboards/minivan/keymaps/josjoha/keymap.c

```diff
@@ -0,0 +1,1187 @@
+    _HALF_,   // BASE layer is _DDL, jumps to 'normal' Unicode maps
+    _FULL_,   // BASE layer is _DDL, jumps to special Linux 'descramble' Unicode maps
+};
+// To remember if we are in descramble mode for 'escape'ing out of layers to the right base
+#ifndef STARTUP_ALTERNATE // Startup with no 'descramble', normal mode.
+ short descramble = _NORMAL_;
+#endif
+#ifdef STARTUP_ALTERNATE  // Startup with 'descramble' (which is 'dvorak' with QWERTY_DVORAK defined) active.
+ short descramble = _FULL_; // Set this to _HALF_ if you want non-Linux layers in 'descramble' upon startup.
+#endif
+                      // There are three modes: 0 for everything normal, 1 for descramble for letters and number/symbols,
+                      // .. but with the normal unicode layers, and 2 for all in descramble mode, where the Unicode
+                      // .. coding is for a Linux system (shift-control-U + HEX input).
+                      // The 1 mode is added, for systems where the Unicode input might work the same whether or not
+                      // .. the computer is already set to Dvorak. (This is untested as far as the result on  those
+                      // .. systems XXX.)
+
+//* Shift detection
+bool shift_ison = 0; // keep track of the state of shift (Capslock is ignored). There may be more elegant code for this in
+                     //   QMK (a function seems to do it?), but this is simple and keeps the issue isolated to this file.
+#define TRUE 1
+#define FALSE 0
+bool _fun_stay = FALSE; // for making _FUN layer not return to BASE after pressing f-key
+bool leds_on = TRUE; // toggle leds on/off
+bool isolate_trigger = FALSE; // detects if _FUN layer shift was pressed and no other key 
+
+
+/* This file contains mostly the Unicode and special macros.
+   It contains the function: process_record_user(...)
+   It has been isolated because the source file got long.
+ */
+#include "./unicode_macros.c"
+
+
+// pre-existing function
+void keyboard_post_init_user (void) {
+  #ifdef RGBLIGHT_ENABLE
+    // Set up RGB effects on _only_ the first LED 
+    rgblight_set_effect_range (1, 1); // Takes a range: 1st arg is start, 2nd how many
+    rgblight_sethsv_noeeprom (HSV_WHITE); // Startup color of keyboard.
+    // Set LED effects to breathing mode
+    rgblight_mode_noeeprom (RGBLIGHT_EFFECT_BREATHING + 2);
+
+    // Init the first and last LEDs to a static color.
+    setrgb (0, 0, 0, (LED_TYPE *)&led[0]); // Led[0] is led 0
+    setrgb (0, 0, 0, (LED_TYPE *)&led[2]); // 2nd led
+    rgblight_set ();
+  #endif //RGBLIGHT_ENABLE
+
+// Set startup layer
+#ifndef STARTUP_ALTERNATE 
+    activate_this_layer (_LTR); // Startup layer
+    deactivate_all_but (_LTR);
+#endif
+#ifdef STARTUP_ALTERNATE 
+    activate_this_layer (_DDL); // Startup layer
+    deactivate_all_but (_DDL);
+#endif
+}
+
+
+// De-activates all other layers than the one given. Does not activate the one given.
+void deactivate_all_but (int layer) {
```

Comment:
This is completely redundant.  `layer_move` does this, by turning off all layers, and then turning on just the specified layer.